lang/go/bindings/examples/distributor/pkg/distributor/rx.go

```diff
@@ -0,0 +1,79 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2022-2023 Intel Corporation
+ */
+
+package distributor
+
+import (
+	"context"
+	"sync"
+
+	"github.com/CloudNativeDataPlane/cndp/lang/go/bindings/cne"
+)
+
+type RX struct {
+	handle *cne.System
+	ctx    context.Context
+
+	lb LoadBalancer
+	wg sync.WaitGroup
+}
+
+func NewRX(ctx context.Context, handle *cne.System, lb LoadBalancer) *RX {
+	rx := &RX{
+		ctx:    ctx,
+		handle: handle,
+		lb:     lb,
+	}
+
+	return rx
+}
+
+func (rx *RX) receive(thdName string, lportNames []string) {
+	lports := rx.handle.LPortsByName(lportNames)
+	if len(lports) == 0 {
+		return
+	}
+
+	err := rx.handle.RegisterThread(thdName)
+	if err != nil {
+		return
+	}
+	defer rx.handle.UnregisterThread(thdName)
+
+	packets := make([]*cne.Packet, numPackets)
+
+	var lportIds []int
+	for _, lport := range lports {
+		lportIds = append(lportIds, lport.LPortID())
+	}
+
+	rx.wg.Add(1)
+	for {
+		for _, pid := range lportIds {
+			select {
+			case <-rx.ctx.Done():
+				rx.wg.Done()
+				return
+			default:
+				size := cne.RxBurst(pid, packets)
+				for i := 0; i < size; {
+					i += rx.lb.Handle(packets[i])
```

Comment:
Does the handler function only return a one, it seems odd to have the handler return the increment value for the loop. It seems like the handler should only return an error or nothing. What happens if the handler returns zero, it can be stuck in this loop. I have not looked at the handler routine yet, so returning zero may not be an option.

Reply:
`Handle` returns how many packets (1 or 0 in this case) has been distribute to workers, just like `EnqueueN(packet, 1)`